vent/controller/control_module.py

```diff
@@ -804,11 +836,39 @@ def _set_HAL(self, valve_open_in, valve_open_out):
     # @timeout
     def _get_HAL(self):
         """
-        Get sensor values from HAL, decorated with timeout
+        Get sensor values from HAL, decorated with timeout.
         """
-        self._DATA_Qout = 0 #self.HAL.flow_ex                      # Flow sensor on Expiratory side   NOTE: THIS DOESNT WORK RN!
-        self._DATA_Qin  = 0# self.HAL.flow_in                      # Flow sensor on inspiratory side. NOTE: used to calculate VTE
-        self._DATA_PRESSURE = self.HAL.pressure
+        glitchcatcher = True
+
+        if not glitchcatcher:
+            self._DATA_PRESSURE = self.HAL.pressure
+            self._DATA_Qout     = self.HAL.flow_ex
+            self._DATA_OXYGEN   = self.HAL.oxygen
+        
+        else:
+            pp = self.HAL.pressure
+            # if np.abs( pp  - self._DATA_PRESSURE ) < 5: # This is not a glitch, save it
```

Comment:
@zajdel trieed to catch the glitches here with these "if". Do you have a better idea?

Reply:
added to #115 as a glitch. Guess we have to wait for now for the revised HAL